### ISSUE TITLE
Reproducing results from paper

### ISSUE THREAD

**Issue by PiotrKaszuba** [zetayue/MXMNet#12]
Dear Authors,
I am trying to reproduce results claimed by your paper and I'm failing to do so. My goal is to reproduce the best results with multiple targets (best config per target). I have managed to reproduce the results (1.19 and 1.16) on target 6 (ZPVE) and only on this one. Currently, I am stuck trying to reproduce the result for target 2. I have run the code available in repository as well as in my refactored version. I've tried multiple options: 

- standard run provided by this repository (python -u main.py --target 2 --cutoff 10 --batch_size 128)
- run with my fork (https://github.com/PiotrKaszuba/MXMNet) with additional molecules removed found in https://github.com/yifeiwang15/MotifConv/blob/main/MCM_for_qm9/excludedMolecules_MG.txt
- run with DimeNet style split also found in MotfConv repo: https://github.com/yifeiwang15/MotifConv/blob/c350e307a16f4ae8e517546cf8263531f77888cd/MCM_for_qm9/main.py#L129
- run with old values of local RBF and SBF modules cutoffs of 3.6179 (https://github.com/zetayue/MXMNet/blob/c96bcaa9d2c918ba313de8819c3fb49e1a5ecb5e/model.py#L32-L34)
- run with current code but swapped data to more accurate (than those provided by QM9 downloader - more decimal places for positions) XYZ positions of atoms and targets found in intial version of repo: https://github.com/zetayue/MXMNet/tree/c96bcaa9d2c918ba313de8819c3fb49e1a5ecb5e/data/QM9/raw/QM9_node_attributes.txt and QM9_graph_attributes.txt (graph_attributes, after applying relevant transformations were the same though; for most targets)
- as well as multiple repetitions of some of above options

For all these trials I am getting test set error ranging from 26 - 25 while in the paper it is 22.8. However, I am always using the default seed of 920 assuming that it was used for your experiments.

Could you help me by providing some instructions or advice how to reproduce the result of 22.8 for target 2, and ideally, for other targets too. For confirmation:

- The seed - is it 920 or different for different targets?,
- The QM9 version - is it the one automatically downloaded by the code?
- Dataset size: 130462 with additional excluded molecules (mentioned above) or 130831 with molecules excluded by the original code (uncharacterized.txt file),
- (Whether or not dictated by the seed) the dataset split - i.e. molecule ids for train/val/test, same for all targets or different? - I can also provide my split for you to confirm if providing molecule ids is too much.

And any other ideas / hints that I could be missing to reproduce the results.

Thank you for your time reading this issue and any assistance you can offer!

**Comment by zetayue** [comment]
Hi,

Thank you for your interest in our paper and your efforts in conducting the experiments!

The default learning rate (lr) provided is 1e-4 in our code (https://github.com/zetayue/MXMNet/blob/master/main.py#L24), while the actual lr on QM9 is chosen from 1e-3 or 1e-4 as suggested in Table 4 of our paper (https://arxiv.org/pdf/2011.07457.pdf). 

We use lr=1e-3 for dipole moment, HOMO, LUMO, HOMO-LUMO gap, and R2, while lr=1e-4 for the other targets. In general, using a larger lr results in a faster decreasing speed of MAE than using a smaller one. However, it is not guaranteed that a lower MAE can be reached after convergence when using a larger lr.

> - The seed - is it 920 or different for different targets?

The seed would not affect the MAE much. Using the default seed should get close results as listed in our paper.

> - The QM9 version - is it the one automatically downloaded by the code?

Yes, the automatically downloaded one is the standard one to be used.

> - Dataset size: 130462 with additional excluded molecules (mentioned above) or 130831 with molecules excluded by the original code (uncharacterized.txt file),

We use the 130462 version following previous works like MPNN, SchNet, PhysNet, MEGNet, MGCN, etc.

> - (Whether or not dictated by the seed) the dataset split - i.e. molecule ids for train/val/test, same for all targets or different? - I can also provide my split for you to confirm if providing molecule ids is too much.

The data split should be affected by the seed value. In each run of our experiment, we use the same seed value to predict all targets.

Overall, using the current code with a proper lr and excluding the molecules in QM9 would be helpful. Let me know if you still meet any issues!

**Comment by PiotrKaszuba** [comment]
Thank you for the reply, answering my questions, and especially for pointing out the learning rate value of 1e-3! I've naively assumed the best config applied to all targets or the same config was chosen across all targets as best on average - and that it was then reflected by the code. I've already applied your advice to target 2 (HOMO), with dataset size of 130462, default seed using such config: --target 2 --cutoff 10 --batch_size 128 --lr 1-e3, and was able to obtain (as of now, model is still training) error of 22.6 on the test data!

>> The seed - is it 920 or different for different targets?

> The seed would not affect the MAE much. Using the default seed should get close results as listed in our paper.

I found it also interesting - I'd suspect that the data splits have matching 'distributions' because of the substantial size of QM9 compared to the complexity of the molecules.

 Thank you again!
